sdk/python/tests/test_client.py

```diff
@@ -400,130 +409,125 @@ def test_stop_ingest_job(self, mocked_client, mocker):
-    #                 file_uris=[f"file://{final_results}"],
-    #                 data_format=DataFormat.DATA_FORMAT_AVRO,
-    #             )
-    #         ),
-    #     )
-    #
-    #     mocker.patch.object(
-    #         mocked_client._serving_service_stub,
-    #         "GetJob",
-    #         return_value=GetJobResponse(
-    #             job=BatchFeaturesJob(
-    #                 id="123",
-    #                 type=JobType.JOB_TYPE_DOWNLOAD,
-    #                 status=JobStatus.JOB_STATUS_DONE,
-    #                 file_uris=[f"file://{final_results}"],
-    #                 data_format=DataFormat.DATA_FORMAT_AVRO,
-    #             )
-    #         ),
-    #     )
-    #
-    #     mocker.patch.object(
-    #         mocked_client._serving_service_stub,
-    #         "GetFeastServingInfo",
-    #         return_value=GetFeastServingInfoResponse(
-    #             job_staging_location=f"file://{tempfile.mkdtemp()}/",
-    #             type=FeastServingType.FEAST_SERVING_TYPE_BATCH,
-    #         ),
-    #     )
-    #
-    #     mocked_client.set_project("project1")
-    #     response = mocked_client.get_batch_features(
-    #         entity_rows=pd.DataFrame(
-    #             {
-    #                 "datetime": [
-    #                     pd.datetime.now(tz=timezone("Asia/Singapore")) for _ in range(3)
-    #                 ],
-    #                 "customer": [1001, 1002, 1003],
-    #                 "transaction": [1001, 1002, 1003],
-    #             }
-    #         ),
-    #         feature_refs=[
-    #             "my_project/customer_feature_1",
-    #             "my_project/customer_feature_2",
-    #         ],
-    #     )  # type: Job
-    #
-    #     assert response.id == "123" and response.status == JobStatus.JOB_STATUS_DONE
-    #
-    #     actual_dataframe = response.to_dataframe()
-    #
-    #     assert actual_dataframe[
-    #         ["my_project/customer_feature_1", "my_project/customer_feature_2"]
-    #     ].equals(
-    #         expected_dataframe[
-    #             ["my_project/customer_feature_1", "my_project/customer_feature_2"]
-    #         ]
-    #     )
+    @pytest.mark.parametrize(
+        "mocked_client",
+        [pytest.lazy_fixture("mock_client"), pytest.lazy_fixture("secure_mock_client")],
+    )
+    def test_get_batch_features(self, mocked_client, mocker):
+
+        mocked_client._serving_service_stub = Serving.ServingServiceStub(
+            grpc.insecure_channel("")
+        )
+        mocked_client._core_service_stub = Core.CoreServiceStub(
+            grpc.insecure_channel("")
+        )
+
+        mocker.patch.object(
+            mocked_client._core_service_stub,
+            "GetFeatureSet",
+            return_value=GetFeatureSetResponse(
+                feature_set=FeatureSetProto(
+                    spec=FeatureSetSpecProto(
+                        name="driver",
+                        project="driver_project",
```

Comment:
Would it make sense to have an online and batch test where no project is defined? Unless that is being tested elsewhere.

Reply:
That should be tested in the e2e tests